### PR TITLE
Added an StreamQueryExplainer interface for returning additional info…

### PR DESCRIPTION
**Pull request by kodjo-anipah**
Adding a StreamQueryExplainer interface to the Graylog2 server, for analyzing whether queries, would overlap with data located in a data-routed backend. 
 Extended ValidationResponseDTO.java 
 *  include the absolute time range requested to ease the retrieval of data.
 * extended context with StreamRoutedData which holds information about the stream and the routed data destination

relates to https://github.com/Graylog2/graylog-plugin-enterprise/issues/7821
/nocl

### CODE REVIEW

File: graylog2-web-interface/src/views/components/searchbar/queryvalidation/QueryValidation.tsx [hunk 1]
```diff
@@ -190,24 +195,25 @@ const QueryValidation = () => {
...
         <Container ref={explanationTriggerRef}>
           {hasExplanations ? (
-            <ExplanationTrigger title="Toggle validation error explanation"
+            <ExplanationTrigger title={`Toggle validation ${isInfo ? 'information' : 'error explanation'}`}
                                 onClick={toggleShow}
                                 $clickable
                                 tabIndex={0}
                                 type="button">
-              <ErrorIcon $status={status} name="error" />
+              {isInfo && <ErrorIcon $status={status} name="database" />}
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/searchbar/queryvalidation/QueryValidation.tsx:215]: What about this:
```suggestion
              <ErrorIcon $status={status} name={isInfo ? "database" : "error"} />
```

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java [hunk 2]
```diff
@@ -79,17 +89,34 @@ public ValidationResponseDTO validateQuery(
...
+                dataRoutedStreams,
+                requestedRangeAsAbsolut);
     }
 
     private boolean containsWarmIndices(Set<ExplainResults.IndexRangeResult> searchedIndexRanges) {
         return searchedIndexRanges.stream().anyMatch(ExplainResults.IndexRangeResult::isWarmTiered);
     }
 
+    Set<DataRoutedStream> checkForDataRoutedStreams(Optional<Set<String>> requestedStreams, Set<String> readableStreams, TimeRange timeRange, boolean isEmptyQuery) {
+        if (optionalStreamQueryExplainer.isPresent()) {
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java:111]: Minor nitpick, we could also do it like this:
```suggestion
        return optionalStreamQueryExplainer.ifPresent(explainer -> { ... }).orElse(Set.of());
```

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good! Just two minor suggestions for improvement inline.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java (hunk 2 above)
Comment (kodjo-anipah) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java]: > Minor nitpick, we could also do it like this:

right I can use optional.map to do that

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java]: `Optional#map` right, corrected that. :+1:

**Review by dennisoelkers — APPROVED** [review]

**Pull request merged**